### DISCUSSION TITLE
why start_date is a required argument to dag

### DISCUSSION THREAD

**Discussion by aamster** [apache/airflow#28903] [Q&A] [answered]
My use case is that I want the scheduler to only schedule a workflow run when manually triggered. So I set `schedule` to `None`.

In this setting `start_date` and `backfill` doesn't make sense. Why is it required?

**Comment by potiuk** [comment, marked as answer]
Start_date mostly reflects the time when given dag started to exist. For scheduled dags with "catchup=True" it means that it will backfill the past automatically (see the other question for backfill) For manual dags it's mostly meta-data for  the organisational reason. You simply stay - this dag has been really valid since then. Maybe it's not needed per-se for scheduling but for consistenty and tracking and manageability of many dags, it is still useful.

You can call it "decorative" in this case. I think that's the purpose of how I understand it. 

I think also a bit historical and possibly it could be changed if you think it makes a lot of sense - this is a bit fundamental change and would require a deliberate proposal, discussion, and POC (Proof of Concept) published in PR and discussed in the devlist (see https://airflow.apache.org/community/) but if you would like to proceed with changing it (this is an open-source project done mostly as voluntary contributions)  - you would have to learn quite a lot of internals to attempt to remove the limit - likely a lot of code depends on it. Usually if someone proposes a change here, they take the task on making sure that the proposal is sound , and the burden of analysing and checking that falls on the side that proposes it.

**Reply by aamster** [reply]
It should default to now or be None rather than being required IMO. It doesn’t seem to serve a real purpose in a manually executed workflow use case. But not sure I want to contribute right now. Thanks!

**Reply by potiuk** [reply]
Many things **should be** different when time passes and things change. But only changing them to how they should be, moves a needle. Other than than that it is just whining.
